app/connectors/VatReturnWithCorrectionHttpParser.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import logging.Logging
+import models.corrections.CorrectionPayload
+import models.domain.VatReturn
+import models.responses._
+import play.api.http.Status._
+import play.api.libs.json.{JsError, JsSuccess}
+import uk.gov.hmrc.http.{HttpReads, HttpResponse}
+
+object VatReturnWithCorrectionHttpParser extends Logging {
+
+  type VatReturnWithCorrectionResponse = Either[ErrorResponse, (VatReturn, CorrectionPayload)]
+
+  implicit object VatReturnWithCorrectionReads extends HttpReads[VatReturnWithCorrectionResponse] {
+    override def read(method: String, url: String, response: HttpResponse): VatReturnWithCorrectionResponse = {
+      response.status match {
+        case OK | CREATED =>
```

Comment:
Can we still have OK and created scenarios, I think the previous one had it because we used it for the post and get, does this still apply here?

Reply:
EDIT *I meant to put this comment on the correction Parser

Reply:
Yes, it would be the case here as well so we can reuse